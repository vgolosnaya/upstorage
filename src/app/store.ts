import { configureStore } from '@reduxjs/toolkit';
import filesReducer from './file/files.reducer';

export const store = configureStore({
    reducer: {
        files: filesReducer,
    },
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
