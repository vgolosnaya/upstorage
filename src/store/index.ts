import { combineReducers, configureStore } from '@reduxjs/toolkit';
import filesReducer from 'store/files/files.reducer';
import { IFile } from 'services/FileReader.service';
import filesMiddleware from 'store/files/files.middleware';
import StorageService from 'services/Storage.service';
import errorsReducer, { ErrorState } from 'store/errors/errors.reducer';

const storageService = new StorageService<IFile>('files');

export const store = configureStore({
    reducer: combineReducers({
        files: filesReducer,
        errors: errorsReducer,
    }),
    middleware: [ filesMiddleware(storageService) ]
});

export type AppDispatch = typeof store.dispatch;
export type RootState = {
    files: IFile[],
    errors: ErrorState[],
};
