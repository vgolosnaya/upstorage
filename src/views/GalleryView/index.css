.gallery-container {
    grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
    display: grid;
    gap: 1rem;
    padding: 2rem;
}

.img-item {
    width: 300px;
}