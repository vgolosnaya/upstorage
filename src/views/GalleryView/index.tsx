import React from 'react';
import { useAppSelector } from 'store/hooks';
import { Container, Card, CardMedia, CardContent, Typography } from '@mui/material';
import filesSelectors from 'store/files/files.selectors';
import './index.scss';
import LABELS from '../../labels';

const GalleryView = (): JSX.Element => {
    const images = useAppSelector(filesSelectors.getImages);
    
    return (
        <div className="gallery-container">
            {images.map((file) => (
                <Card key={file.id}>
                    <CardMedia
                        component="img"
                        image={file.data}
                        height="200"
                        alt={file.fileName}
                        sx={{ 'object-fit': 'scale-down' }}
                    />
                    <CardContent>
                        <Typography gutterBottom variant="body2" component="div">
                            {file.fileName}
                        </Typography>
                        <Typography variant="body2" color="text.secondary">
                            {LABELS.uploaded} {file.relativeTimestamp}
                        </Typography>
                    </CardContent>
                </Card>
            ))}
        </div>
    );
};

export default GalleryView;
