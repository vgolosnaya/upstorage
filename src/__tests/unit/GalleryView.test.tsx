import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { store } from 'store';
import React from 'react';
import GalleryView from 'views/GalleryView';

describe('[SNAPSHOT] Gallery page:', ()=>{
    test('Should match snapshot', () => {
        const component = render(
            <Provider store={store}>
                <GalleryView/>
            </Provider>
        );
        
        expect(component.baseElement).toMatchSnapshot();
    });
});