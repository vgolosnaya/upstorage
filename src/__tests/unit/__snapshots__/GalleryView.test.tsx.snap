// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[SNAPSHOT] Gallery page: Should match snapshot 1`] = `
<body>
  <div>
    <div
      class="gallery-container"
    >
      <span
        class="MuiTypography-root MuiTypography-overline css-b3fch5-MuiTypography-root"
      >
        You don't have any file loaded
      </span>
    </div>
  </div>
</body>
`;
